webapp/channels/src/packages/mattermost-redux/src/actions/channels.ts

```diff
@@ -1048,6 +1048,47 @@ export function addChannelMember(channelId: string, userId: string, postRootId =
     };
 }
 
+export function addChannelMembers(channelId: string, userIds: string[], postRootId = ''): ActionFuncAsync {
+    const batchSize = 1000;
+    return async (dispatch, getState) => {
+        const batches = [];
+        for (let i = 0; i < userIds.length; i += batchSize) {
+            batches.push(userIds.slice(i, i + batchSize));
+        }
```

Comment:
I don't see what is this being used for.. that said, I think the webapp action should just call this with all the user_ids and the batching should occur here instead of the webapp action, then dispatch all of them together...